nav2_simple_commander/nav2_simple_commander/robot_navigator.py

```diff
@@ -409,6 +423,7 @@ def dockRobotByID(self, dock_id, nav_to_dock=True):
         self.goal_handle = send_goal_future.result()
 
         if not self.goal_handle.accepted:
+            self.setTaskError(DockRobot.UNKNOWN, 'Docking request was rejected')
```

Comment:
Only some `if not self.goal_handle.accepted:` have the `setTaskError`, I think you're missing most of them